From 4be96f98bac3eeac0fad40ca0450fcb7b9d0313d Mon Sep 17 00:00:00 2001
From: Victor Toso <me@victortoso.com>
Date: Tue, 16 May 2017 19:21:21 +0200
Subject: [PATCH 32/55] vdagentd: Do not call session_info_is_user() with
 invalid session_info

If we pass -X command line option which disables
console-kit/systemd-logind integration, we will have the following
critical being issued:

  CRITICAL **: session_info_is_user: assertion 'si != NULL' failed

This patch avoid it.

Signed-off-by: Victor Toso <victortoso@redhat.com>
Acked-by: Pavel Grunt <pgrunt@redhat.com>
---
 src/vdagentd/vdagentd.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/vdagentd/vdagentd.c b/src/vdagentd/vdagentd.c
index f3ac606..201efc4 100644
--- a/src/vdagentd/vdagentd.c
+++ b/src/vdagentd/vdagentd.c
@@ -786,7 +786,9 @@ static void update_active_session_connection(struct udscs_connection *new_conn)
     if (debug)
         syslog(LOG_DEBUG, "%p is now the active session", new_conn);
 
-    if (active_session_conn && !session_info_is_user(session_info)) {
+    if (active_session_conn &&
+        session_info != NULL &&
+        !session_info_is_user(session_info)) {
         if (debug)
             syslog(LOG_DEBUG, "New session agent does not belong to user: "
                    "disabling file-xfer");
-- 
2.16.1

